eventuals/collect.h

```diff
@@ -4,6 +4,7 @@
 
 #include "eventuals/loop.h"
 #include "eventuals/type-traits.h"
+#include "google/protobuf/repeated_field.h"
```

Comment:
This feels like a pretty heavy dependency for users that aren't using protos.

Is there any way we can avoid it, e.g. by splitting out the proto parts into a separate file?

I have no idea how to do this, so if I was writing this PR I'd have this same `#include`, even though it feels messy :/

Maybe add a TODO to figure out a cleaner way to separate out proto overrides from proto-agnostic stuff?

Reply:
I'm not sure if we want a TODO here. It might be better to make an issue for that.

Reply:
![why not both](https://c.tenor.com/odyVsZbC-OYAAAAC/why-not-both-why-not.gif)

I think of TODOs kind of like hyperlinks to issues, plus they give the reader more context (like normal contexts). Personally, I add them liberally :)

Up to you, though.

Reply:
Added a simple note. If we encounter any problems with compilation times while including `eventuals/collect.h` we'll at least see "HERE'S YOUR PROBLEM" thingy :D